### ISSUE TITLE
Avoid repeated license requests when seeking between DASH periods

### ISSUE THREAD

**Issue by sergiomartinez4** [google/ExoPlayer#6725]
Hi ExoPlayer team,

I'm currently playing a Multi-Period Live DASH with Widevine, and I'm noticing that when seeking across periods, a license request is made every time. When letting the stream play and cross period by normal playback, it does not request the license. It also does not request a license when seeking within the same period.
Are multiple requests expected when seeking across periods?

I tested it on the ExoPlayer Demo version 2.10.8.
The bug report and the media information will be sent to dev.exoplayer@gmail.com.
Thanks

**Comment by sergiomartinez4** [comment]
Hello team,
is there an update about this question?
Any info will be much appreciated
Thanks

**Comment by KelvinJin** [comment]
Hey guys, instead of creating a new issue I just want to mention that we're having the same issue here. Please indicate if this is necessary and if there's a way to stop the extra requests. This actually increases the cost of our DRM server so it would be good to figure out a solution. Thanks in advance.

**Comment by icbaker** [comment]
This is currently working as expected (unfortunately) - when seeking from one period to another we release the previous period & renderer (which in turn causes the `DrmSession` to be released) before starting playback in the new period (which triggers acquiring the `DrmSession`).

This means that we're not able to re-use the `DrmSession` instance, even though the keys are identical, because it's been fully released. So we initialize a new one and that causes another key request.

When we transition from one period to the next during normal playback (not seeking) we keep the old renderer around just long enough to keep the `DrmSession` alive until starting to play the new period, so when we acquire the session from the new period we're able to re-use the existing one and avoid another key request.

I'm looking into some different approaches to solve this, but can't promise a timeline I'm afraid.

**Comment by icbaker** [comment]
316f8a8 implements session keepalive inside the `DefaultDrmSessionManager`. By default sessions are kept for 5 minutes after the last 'release' - this can be customised with `DefaultDrmSessionManager.Builder#setSessionKeepaliveMs(long)`. This is (more than!) enough to bridge the gap when seeking between DASH periods - so the session should now be re-used after the seek.

This is available on the dev-v2 branch - please try it out and let us know if you spot any problems!

**Comment by KelvinJin** [comment]
Just want to report back that the patch does fix the issue. And look forward to the release! Thanks again!
